core/src/main/java/org/infinispan/stream/impl/local/AbstractLocalCacheStream.java

```diff
@@ -27,6 +27,8 @@
    protected Set<?> keysToFilter;
    protected boolean parallel;
 
+   private S stream;
```

Comment:
This one shouldn't be required either.

Reply:
This is needed so that calls to close() the stream actually delegate to the lazily created delegate stream.